packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -157,6 +173,27 @@ class ${class_name}:
             result[k - 1] = v
         return Output._make(result)
 
+    def _get_enum_type(self, parameter: Any, enum_values_by_type_name: Dict[str, Dict[str, Any]]) -> Any:
+        if parameter.type == FieldDescriptorProto.TYPE_ENUM:
+            loaded_enum_values = json.loads(parameter.annotations["ni/enum.values"])
+            enum_values = dict((key, value) for key, value in loaded_enum_values.items())
+
+            for existing_enum_name, existing_enum_values in enum_values_by_type_name.items():
+                if existing_enum_values == enum_values:
+                    return Enum(existing_enum_name, existing_enum_values)
+
+            new_enum_type_name = self._get_enum_class_name(parameter.name)
+            enum_values_by_type_name[new_enum_type_name] = enum_values
+            return Enum(new_enum_type_name, enum_values)
+        return None
+
+    def _get_enum_class_name(self, name: str) -> str:
+        class_name = name.title().replace(" ", "")
+        invalid_chars = "`~!@#$%^&*()-+={}[]\|:;',<>.?/ \n_"
+        pattern = f"[{re.escape(invalid_chars)}]"
+        class_name = re.sub(pattern, '', class_name)
+        return class_name + "Enum"
```

Comment:
> Maybe it's fine to turn camelCase and PascalCase into garbage, though? 

I wanted to create Enums in Pascal case based on other enums' naming conventions in the repository. Since `title()` doesn't work, I've created a method that filters out illegal characters, splits the string, and capitalizes each segment. This will convert any case inputs to Pascal case by excluding illegal characters. 
Since the enum class name is derived from measurement configuration or output parameter name, I hope we don't have to look for rare unusual cases. Please let me know your thoughts on this.

> Why are you creating enum classes at run time, anyway? The generated code already defines the enum classes in code. 

The generated code defines the enum classes, however, when using the client APIs, in `_create_file_descriptor`, we might not know the enum type for a particular enum parameter. Thus, we need to get the enum types at runtime also. 
At this stage, `configuration` is of type `measurement_service_pb2.ConfigurationParameter`, so we won't be having `enum_type` attribute in it.

Reply:
> Please let me know your thoughts on this.

My thoughts are that we are not giving users enough guidance on naming style, and this makes our lives harder when generating client code. Please make sure your team updates the documentation to offer better guidance.

As for the details: I'll look at those on Monday.

> The generated code defines the enum classes, however, when using the client APIs, in _create_file_descriptor, we might not know the enum type for a particular enum parameter. Thus, we need to get the enum types at runtime also.
At this stage, configuration is of type measurement_service_pb2.ConfigurationParameter, so we won't be having enum_type attribute in it.

Great, we have the information but we're not using it. Please structure the code to make this information available where it's needed, instead of regenerating it by using the Python enum functional API in a way that Mypy flags as incorrect usage.

Reply:
> we have the information but we're not using it. 

Yes! We missed this. We already have the data in` self._configuration_metadata` and `self._output_metadata`. While `_create_file_descriptor` instead of using the above data, we try to get the data using `GetMetadata`. So, I have removed the code where we try to get the metadata and used the already available information from `self._configuration_metadata` and `self._output_metadata`